Week1/homework/app.js

```diff
@@ -1,11 +1,149 @@
+    title: 'To Miss With Love',
+    language: 'English',
+    author: 'Katharine Birbalsingh'
+  },
+  where_the_wild_things_are: {
+    title: 'Where the wild Things are',
+    language: 'English',
+    author: 'Maurice Sendak'
+  },
+  me_talk_pretty_one_day: {
+    title: 'Me Talk Pretty One Day',
+    language: 'English',
+    author: 'David Sedaris'
+  },
+  another_bullshit_night_in_suck_city: {
+    title: 'Another Bullshit Night in Suck City',
+    language: 'English',
+    author: 'Nick Flynn'
+  }
+};
+
+//  1.5------1.8
+function getBooksInfo() {
+  const ulDom = document.createElement('ul');
+  document.body.appendChild(ulDom);
+  ulDom.setAttribute('class', 'books');
+  for (let i = 0; i < bookTitles.length; i++) {
+    // const divDom1 = document.createElement('div');
+    const liDom = document.createElement('li');
+    const divDom2 = document.createElement('div');
+    const divDom3 = document.createElement('div');
+    const h1Dom = document.createElement('h1');
+    const h4Dom = document.createElement('h4');
+    const h5Dom = document.createElement('h5');
+    const img = document.createElement('img');
+
+    divDom2.setAttribute('class', 'book_info');
+    divDom3.setAttribute('class', 'book_cover');
+    liDom.setAttribute('id', bookTitles[i] + '_container');
+    liDom.setAttribute('class', 'book');
+    h1Dom.setAttribute('class', 'books_titles');
+    h4Dom.setAttribute('class', 'book_author');
+    h4Dom.setAttribute('id', bookTitles[i] + '_author');
+    h5Dom.setAttribute('class', 'book_language');
+    h5Dom.setAttribute('id', bookTitles[i] + '_language');
+    img.setAttribute('src', 'images/' + bookTitles[i] + '.jpg');
+    img.setAttribute('alt', bookListObj[`${bookTitles[i]}`] + ' cover');
+    img.setAttribute('id', bookTitles[i]);
+
+    h1Dom.textContent = bookListObj[`${bookTitles[i]}`].title;
+    h4Dom.textContent = bookListObj[`${bookTitles[i]}`].author;
+    h5Dom.textContent = bookListObj[`${bookTitles[i]}`].language;
+
+    ulDom.appendChild(liDom);
+    liDom.appendChild(divDom3);
+    divDom3.appendChild(divDom2);
+    divDom3.appendChild(img);
+    divDom2.appendChild(h1Dom);
+    divDom2.appendChild(h4Dom);
+    divDom2.appendChild(h5Dom);
+  }
+}
+getBooksInfo();
+
+function changeBg() {
+  for (let i = 0; i < bookTitles.length; i++) {
+    const img = document.getElementById(bookTitles[i]);
+    img.addEventListener('mouseover', function() {
+      img.setAttribute('src', 'images/' + bookTitles[i] + '_author.jpg');
+      img.setAttribute('alt', bookListObj[`${bookTitles[i]}`] + ' author');
+    });
+  }
+}
+changeBg();
+function changeBgOriginal() {
+  for (let i = 0; i < bookTitles.length; i++) {
+    const img = document.getElementById(bookTitles[i]);
+    img.addEventListener('mouseout', function() {
+      img.setAttribute('src', 'images/' + bookTitles[i] + '.jpg');
+      img.setAttribute('alt', bookListObj[`${bookTitles[i]}`] + ' Cover');
+    });
+  }
 }
+changeBgOriginal();
```

Comment:
Nice use of event listeners to handle the `mousover` and `mouseout` events. But what if you don't have a mouse, e.g. on a mobile phone or iPad?

The two functions `changeBg` and `changeBgOriginal` both loop through the same `bookTitles` array. You could combine them in one function. Also, it is best to execute all code when the page is fully loaded, using `window.onload`, for example:


```js
// ...

function addEventListeners() {
  for (const bookTitle of bookTitles) {
    const img = document.getElementById(bookTitle);
    img.addEventListener('mouseover', function() {
      img.setAttribute('src', 'images/' + bookTitle + '_author.jpg');
      img.setAttribute('alt', bookTitle + ' author');
    });
    img.addEventListener('mouseout', function() {
      img.setAttribute('src', 'images/' + bookTitle + '.jpg');
      img.setAttribute('alt', bookTitle + ' Cover');
    });
  }
}

function main() {
  getBooksInfo();
  addEventListeners();
}

window.onload = main;
```